src/Chainweb/BlockHeader.hs

```diff
@@ -374,16 +378,40 @@ epochStart ph@(ParentHeader p) adj (BlockCreationTime bt)
     --    b) adjusting timeBlocked in both directions, possibly based off the
     --       avg?
     --
-    timeBlocked = maximum adjCreationTimes .-. _blockCreationTime p
+    _adjustmentMax = maximum adjCreationTimes .-. _blockCreationTime p
         -- the maximum is at least @_blockCreationTime p@ and thus the result is
         -- greater or equal 0.
 
+    -- This computes @mean adjCreationTimes - _blockCreationTime p
+    --
+    -- It holds that
+    --
+    -- \(\left(mean_{0 \leq i < n} a_i\right)
+    -- = \frac{\sum_{0 \leq i < n} a_i}{n} - t
+    -- = \frac{\left(sum_{0 \leq i < n} a_i\right) - \left(\sum_{0 \leq i < n} t\right)}{n}
+    -- = \frac{sum_{0 \leq i < n} (a_i - t)}{n}
+    -- \)
+    --
+    --
+    -- this is numberically sound because we compute the differences on integral
+    -- types without rounding.
+    --
+    _adjustmentAvg = x `divTimeSpan` length adjCreationTimes
+      where
+        x :: TimeSpan Micros
+        x = foldr1 addTimeSpan $ (.-. _blockCreationTime p) <$> adjCreationTimes
```

Comment:
`foldr` so it can be rewritten and inlined. `foldr1` struggles with this.

Reply:
I can take a look. Generally, this is not performance critical. It's just a single digit number of items and called during DA, i.e. once every 120 blocks.

Reply:
A okay